crypto/ocsp/ocsp_client.c

```diff
@@ -30,3 +30,98 @@ OCSP_BASICRESP *OCSP_response_get1_basic(OCSP_RESPONSE *resp) {
   return ASN1_item_unpack(rb->response, ASN1_ITEM_rptr(OCSP_BASICRESP));
 }
 
+OCSP_SINGLERESP *OCSP_resp_get0(OCSP_BASICRESP *bs, size_t idx)
+{
+  if (bs == NULL) {
+    OPENSSL_PUT_ERROR(OCSP, ERR_R_PASSED_NULL_PARAMETER);
+    return NULL;
+  }
+  return sk_OCSP_SINGLERESP_value(bs->tbsResponseData->responses, idx);
+}
+
+int OCSP_resp_find(OCSP_BASICRESP *bs, OCSP_CERTID *id, int last)
+{
+  if (bs == NULL || id == NULL){
+    OPENSSL_PUT_ERROR(OCSP, ERR_R_PASSED_NULL_PARAMETER);
+    return -1;
+  }
+
+  STACK_OF(OCSP_SINGLERESP) *sresp = bs->tbsResponseData->responses;
+  OCSP_SINGLERESP *single;
+
+  if (last < 0) {
+    last = 0;
+  } else {
+    last++;
+  }
+  for (size_t i = last; i < sk_OCSP_SINGLERESP_num(sresp); i++) {
+    single = sk_OCSP_SINGLERESP_value(sresp, i);
+    if (!OCSP_id_cmp(id, single->certId)) {
```

Comment:
`single` can be `NULL`?

Reply:
I assume that since we use the number of OCSP_SINGLERESP returned to loop through the stack, we won't run into a NULL here? (Stacks of OCSP_SINGLERESP are extracted from OCSP_BASICRESP, which are extracted directly from the OCSP_RESPONSE der)